base/system/smss/sminit.c

```diff
@@ -1819,9 +1819,9 @@ SmpCreateDynamicEnvironmentVariables(VOID)
     if (NT_SUCCESS(Status))
     {
         /* To combine it into a single string */
-        swprintf((PWCHAR)PartialInfo->Data + wcslen((PWCHAR)PartialInfo->Data),
-                 L", %S",
-                 PartialInfo2->Data);
+        ((PWCHAR)(PartialInfo->Data))[PartialInfo->DataLength / sizeof(WCHAR)] = L'\0';        
+        wcsncat((PWCHAR)PartialInfo->Data, L", ", 2 * sizeof(WCHAR));
+        wcsncat((PWCHAR)PartialInfo->Data, (PWCHAR)PartialInfo2->Data, PartialInfo2->DataLength / sizeof(WCHAR));		
```

Comment:
So the alternative:
```suggestion
        swprintf(((PWCHAR)PartialInfo->Data) + PartialInfo->DataLength / sizeof(WCHAR),
                 L", %.*s",
                 PartialInfo2->DataLength / sizeof(WCHAR), PartialInfo2->Data);
```
would not work?

Reply:
Does not work, only 1 string is displayed : 
![image](https://user-images.githubusercontent.com/24843587/80695535-57451100-8ad6-11ea-9766-d600d061cbfe.png)